packages/opentelemetry-sdk-node/src/sdk.ts

```diff
@@ -0,0 +1,165 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { HttpTextPropagator, metrics } from '@opentelemetry/api';
+import { ContextManager } from '@opentelemetry/context-base';
+import { MeterConfig, MeterProvider } from '@opentelemetry/metrics';
+import { NodeTracerConfig, NodeTracerProvider } from '@opentelemetry/node';
+import { detectResources, Resource } from '@opentelemetry/resources';
+import { BatchSpanProcessor, SpanProcessor } from '@opentelemetry/tracing';
+import { NodeSDKConfiguration } from './types';
+
+/** This class represents everything needed to register a fully configured OpenTelemetry Node.js SDK */
+export class NodeSDK {
+  private _tracerProviderConfig?: {
+    tracerConfig: NodeTracerConfig;
+    spanProcessor: SpanProcessor;
+    contextManager?: ContextManager;
+    httpTextPropagator?: HttpTextPropagator;
+  };
+  private _meterProviderConfig?: MeterConfig;
+
+  private _resource: Resource;
+
+  private _autoDetectResources: boolean;
+
+  /**
+   * Create a new NodeJS SDK instance
+   */
+  public constructor(configuration: Partial<NodeSDKConfiguration> = {}) {
+    this._resource = configuration.resource ?? new Resource({});
+
+    this._autoDetectResources = configuration.autoDetectResources ?? true;
+
+    if (configuration.spanProcessor || configuration.traceExporter) {
+      const tracerProviderConfig = {} as NodeTracerConfig;
+
+      if (configuration.defaultAttributes) {
+        tracerProviderConfig.defaultAttributes =
+          configuration.defaultAttributes;
+      }
+      if (typeof configuration.logLevel === 'number') {
+        tracerProviderConfig.logLevel = configuration.logLevel;
+      }
+      if (configuration.logger) {
+        tracerProviderConfig.logger = configuration.logger;
+      }
+      if (configuration.plugins) {
+        tracerProviderConfig.plugins = configuration.plugins;
+      }
+      if (configuration.sampler) {
+        tracerProviderConfig.sampler = configuration.sampler;
+      }
+      if (configuration.traceParams) {
+        tracerProviderConfig.traceParams = configuration.traceParams;
+      }
+
+      const spanProcessor =
+        configuration.spanProcessor ??
+        new BatchSpanProcessor(configuration.traceExporter!);
+
+      this.configureTracerProvider(
+        tracerProviderConfig,
+        spanProcessor,
+        configuration.contextManager,
+        configuration.httpTextPropagator
+      );
+    }
+
+    if (configuration.metricExporter) {
+      const meterConfig: MeterConfig = {};
```

Comment:
it will be the same here, except those checks for typeof === 'number' are the rest needed ?